commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -0,0 +1,161 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.android.volley.Cache;
+import com.android.volley.NetworkResponse;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.JsonRequest;
+import com.android.volley.toolbox.Volley;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+
+import java.io.UnsupportedEncodingException;
+
+public class MwVolleyApi {
+
+    private static RequestQueue REQUEST_QUEUE;
+    private static final Gson GSON = new GsonBuilder().create();
+    private Context context;
+
+    public MwVolleyApi(Context context) {
+        this.context = context;
+    }
+    public void request(String apiUrl) {
+        JsonRequest<QueryResponse> request = new QueryRequest(apiUrl,
+                new LogResponseListener<QueryResponse>(), new LogResponseErrorListener());
+        getQueue().add(request);
+    }
+
+    private RequestQueue getQueue() {
+        return getQueue(context);
+    }
+
+    private static RequestQueue getQueue(Context context) {
+        if (REQUEST_QUEUE == null) {
+            REQUEST_QUEUE = Volley.newRequestQueue(context.getApplicationContext());
+        }
+        return REQUEST_QUEUE;
+    }
+
+    private static class LogResponseListener<T> implements Response.Listener<T> {
+        private static final String TAG = LogResponseListener.class.getName();
+
+        @Override
+        public void onResponse(T response) {
+            Log.d(TAG, response.toString());
+        }
+    }
+
+    private static class LogResponseErrorListener implements Response.ErrorListener {
+        private static final String TAG = LogResponseErrorListener.class.getName();
+
+        @Override
+        public void onErrorResponse(VolleyError error) {
+            Log.e(TAG, error.toString());
+        }
+    }
+
+    private static class QueryRequest extends JsonRequest<QueryResponse> {
```

Comment:
Hm, I'm not sure if we're talking about the same thing here. This is what I get when I try to put this class in its own file. I could change the QueryRequest class to not static, but then the same problem occurs with QueryResponse, etc. And if I remove the 'static' keyword, I have to instantiate objects to use these classes, so they don't work like static classes anymore.

![static_class](https://cloud.githubusercontent.com/assets/3611199/11987115/83b9dfea-aa3e-11e5-99f9-f950990ccaa8.jpg)